libraries/botbuilder-applicationinsights/src/applicationInsightsTelemetryClient.ts

```diff
@@ -0,0 +1,144 @@
+/**
+ * @module botbuilder-applicationinsights
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { BotTelemetryClient, Severity } from 'botbuilder-core';
+const appInsights = require('applicationinsights');
+const cls = require('cls-hooked');
+const ns = cls.createNamespace('my.request');
+
+// This is the currently recommended work-around for using Application Insights with async/await
+// https://github.com/Microsoft/ApplicationInsights-node.js/issues/296
+// This allows AppInsights to automatically apply the appropriate context objects deep inside the async/await chain.
+const correlationContextManager = require('applicationinsights/out/AutoCollection/CorrelationContextManager').CorrelationContextManager;
+const origGetCurrentContext = correlationContextManager.getCurrentContext;
+
+function getCurrentContext(){
+  return ns.get('ctx') || origGetCurrentContext();
+}
+
+// Overwrite the built-in getCurrentContext() method with a new one.
+correlationContextManager.getCurrentContext = getCurrentContext;
+
+export const ApplicationInsightsWebserverMiddleware = function (req, res, next) {
+
+  // Check to see if the request contains an incoming request.
+  // If so, set it into the Application Insights context.
+  const activity = req.body;
+  if (activity && activity.id) {
+    const context = appInsights.getCorrelationContext();
+    context.activity = req.body;
+  }
+
+  ns.bindEmitter(req);
+  ns.bindEmitter(res);
+  ns.run(function(){
+    ns.set('ctx', origGetCurrentContext());
+    next();
+  });
+
+};
+
+
+/* ApplicationInsightsTelemetryClient Class
+ * This is a wrapper class around the Application Insights node client.
+ * This is primarily designed to be used alongside the WaterfallDialog telemetry collection.
+ * It provides a pre-configured App Insights client, and wrappers around
+ * the major tracking functions, allowing it to conform to Botbuilder's generic BotTelemetryClient interface.
+ * To use it, create pass in an instrumentation key:
+ * 
+ * ```
+ * const myDialog = new WaterfallDialog('my_dialog', steps);
+ * const appInsightsClient = new ApplicationInsightsTelemetryClient(my_instrumentation_key);
+ * myDialog.telemetryClient = appInsightsClient;
+ * ```
+ */
+export class ApplicationInsightsTelemetryClient implements BotTelemetryClient {
+
+    private _telemetryClient: any;

```

Comment:
Prefixing private variables with an underscore in TS is superfluous since the `private` access modifier identifies them as private

Reply:
Fair point, i've changed these to just be private without underscores.